src/SprykerEco/Zed/Payone/Business/Payment/MethodSender/PayonePartialRefundMethodSender.php

```diff
@@ -0,0 +1,257 @@
+     * @param \Generated\Shared\Transfer\PayoneStandardParameterTransfer $standardParameter
+     * @param \SprykerEco\Zed\Payone\Persistence\PayoneRepositoryInterface $payoneRepository
+     * @param \SprykerEco\Zed\Payone\Persistence\PayoneEntityManagerInterface $payoneEntityManager
+     * @param \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface $standartParameterMapper
+     */
+    public function __construct(
+        AdapterInterface $executionAdapter,
+        PayoneQueryContainerInterface $queryContainer,
+        PaymentMapperReader $paymentMapperReader,
+        PayoneStandardParameterTransfer $standardParameter,
+        PayoneRepositoryInterface $payoneRepository,
+        PayoneEntityManagerInterface $payoneEntityManager,
+        StandartParameterMapperInterface $standartParameterMapper
+    ) {
+        parent::__construct($queryContainer, $paymentMapperManager);
+        $this->executionAdapter = $executionAdapter;
+        $this->standardParameter = $standardParameter;
+        $this->payoneRepository = $payoneRepository;
+        $this->payoneEntityManager = $payoneEntityManager;
+        $this->standartParameterMapper = $standartParameterMapper;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayonePartialOperationRequestTransfer $payonePartialOperationRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\RefundResponseTransfer
+     */
+    public function executePartialRefund(PayonePartialOperationRequestTransfer $payonePartialOperationRequestTransfer): RefundResponseTransfer
+    {
+        $paymentEntity = $this->getPaymentEntity($payonePartialOperationRequestTransfer->getOrder()->getIdSalesOrder());
+        $paymentMethodMapper = $this->getPaymentMethodMapper($paymentEntity);
+        $requestContainer = $paymentMethodMapper->mapPaymentToRefund($paymentEntity);
+
+        $requestContainer->setAmount($payonePartialOperationRequestTransfer->getRefund()->getAmount() * -1);
+        $requestContainer = $this->preparePartialRefundOrderItems($payonePartialOperationRequestTransfer, $requestContainer);
+        $requestContainer = $this->preparePartialRefundExpenses($payonePartialOperationRequestTransfer, $requestContainer);
+        $this->standartParameterMapper->setStandardParameter($requestContainer, $this->standardParameter);
+        $apiLogEntity = $this->initializeApiLog($paymentEntity, $requestContainer);
+
+        $rawResponse = $this->executionAdapter->sendRequest($requestContainer);
+        $responseContainer = new RefundResponseContainer($rawResponse);
+
+        $this->updateApiLogAfterRefund($apiLogEntity, $responseContainer);
+        $this->updatePaymentPayoneOrderItemsWithStatus(
+            $payonePartialOperationRequestTransfer,
+            $this->getPartialRefundStatus($responseContainer)
+        );
+
+        $responseMapper = new RefundResponseMapper();
```

Comment:
* Extract to factory and use as normal dependency.

Reply:
++